src/amp-story-player/amp-story-component-manager.js

```diff
@@ -43,19 +55,39 @@ export class AmpStoryComponentManager {
       return;
     }
 
-    const intersectingCallback = (entries) => {
+    const ioPrerenderCb = (entries) => {
+      entries.forEach((entry) => {
+        if (!entry.isIntersecting) {
+          return;
+        }
+        elImpl.prerenderCallback();
+
+        prerenderObserver.unobserve(elImpl.getElement());
+      });
+    };
+
+    const rootMarginInPercentage =
+      (MIN_VIEWPORT_PRERENDER_DISTANCE * 100).toString() + '%';
+    const prerenderObserver = new IntersectionObserver(ioPrerenderCb, {
+      rootMargin: rootMarginInPercentage,
+    });
+    prerenderObserver.observe(elImpl.getElement());
+
+    const ioLayoutCb = (entries) => {
```

Comment:
I think it might make things a bit easier down the line if we allow `layoutCallback` to be called without `prerenderCallback`. The fact that we HAVE to call prerender before layout is counter intuitive because unlike the rest of AMP, wdyt?

Reply:
Yep sure, made the change to not block layout on prerender. PTAL.